brave/brave-core
# Refactor brave_wallet_provider_delegate_impl to be shared with android

Resolves https://github.com/brave/brave-browser/issues/19393

This PR shares brave_wallet_provider_delegate_impl.cc between desktop and android, and moves platform-dependent UI calls into brave_wallet_provider_delegate_impl_helper.cc.

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
Done a basic test to verify no obvious regression in desktop:
1. Clone https://github.com/bbondy/eth-manual-tests
2. python -m SimpleHTTPServer 8081
3. Load up http://localhost:8081/request.html in brave
4. Click `ethereum.enable` in the test page to connect a local ganache account, and check the account is connected as expected in the panel.
5. Input another ganache account address and click `eth_sendTransaction` in the test page.
6. Verify the transaction is performed as expected.
